xclim/indices/_threshold.py

```diff
@@ -1237,6 +1238,62 @@ def days_with_snow(
     return to_agg_units(out, prsn, "count")
 
 
+@declare_units(tasmax="[temperature]", hurs="[g m-3]", thresh="[temperature]")
```

Comment:
```suggestion
@declare_units(tasmax="[temperature]", hurs="[]", thresh="[temperature]")
```
According to CMIP6, which we follow for variables' meta, the canonical units for hurs are "%", thus it is "dimensionless".

Also, xclim's `convert_units_to` is unable to convert from "mass per volume" to "dimensionless" as requested at line 1272. That's because we haven't implemented this transformation in the 'hydro' context, the one where the water density (1000 kg/m³) is implicit. That can be done in this PR, if you need it.

Reply:
Would it be a lot of trouble to do the transformation? Another option would be to check the hurs data is already in % and not to perform the calculation otherwise.

Reply:
The problem here would be that xclim's `declare_units` is limited to a single dimensionality per variable. A workaround for your suggestion, would be to pass `hurs=None` but that disables the unit check completely, which I find quite inelegant.

Also I'm not sure I understand how "hurs", _relative_ humidity, could be given in "g m-3"? Wouldn't it be _absolute_ humidity in that case?  I would prefer that users convert their data before. We could add examples on how to do that in the docstring or in the doc.

Reply:
After talking to @clairbarnes I have changed the hurs input to dimensionless as in this suggestion.